mobile/examples/question_answering/android/app/src/main/java/ai/onnxruntime/example/question_answering/MainActivity.kt

```diff
@@ -0,0 +1,104 @@
+
+        TitleText = findViewById(R.id.TitleView)
+        ArticleText = findViewById(R.id.ArticleView)
+        AnswerText = findViewById(R.id.AnswerView)
+        QuestionText = findViewById(R.id.QuestionView)
+        DoQAButton = findViewById(R.id.PerformQaButton)
+        // from <The Little Prince>, chapter one, and the first paragraph
+        TitleText?.setText("<The Little Prince>")
+        ArticleText?.setText(
+            "we are introduced to the narrator, a pilot, and his ideas about grown-ups." +
+                    "Once when I was six years old I saw a magnificent picture in a book," +
+                    " called True Stories from Nature, about the primeval forest. " +
+                    "It was a picture of a boa constrictor in the act of swallowing an animal. " +
+                    "Here is a copy of the drawing." +
+                    "In the book it said: \"Boa constrictors swallow their prey whole, " +
+                    "without chewing it. After that they are not able to move," +
+                    " and they sleep through the six months that they need for digestion.\""
+        );
+        QuestionText?.setHint("From which book I saw a magnificent picture?")
+
+        // Initialize Ort Session and register the onnxruntime extensions package that contains the custom operators.
+        // Note: These are used to decode the input image into the format the original model requires,
+        // and to encode the model output into png format
+        val sessionOptions: OrtSession.SessionOptions = OrtSession.SessionOptions()
+        sessionOptions.registerCustomOpLibrary(OrtxPackage.getLibraryPath())
+        ortSession = ortEnv.createSession(readModel(), sessionOptions)
+
+        DoQAButton?.setOnClickListener {
+            try {
+                performQA(ortSession)
+            } catch (e: Exception) {
+                Log.e(TAG, "Exception caught when perform QuestionAnswering", e)
+                Toast.makeText(
+                    baseContext,
+                    "Failed to perform QuestionAnswering",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        ortEnv.close()
+        ortSession.close()
+    }
+
+    private fun updateUI(result: Result) {
+        val default_ans: String = "No answer found, try another question?"
+        AnswerText?.setText(
+            if (result.outputAnswer.equals("[CLS]")) default_ans else result.outputAnswer
+        )
+    }
+
+    private fun readModel(): ByteArray {
+        val modelID = R.raw.csarron_mobilebert_uncased_squad_v2_quant_with_pre_post_processing
+        return resources.openRawResource(modelID).readBytes()
+    }
+
+    private fun readQuestion(): CharSequence {
+        var user_text = QuestionText?.text
+        return if (user_text?.isEmpty() == true) QuestionText?.hint!! else user_text!!
```

Comment:
we might can eliminate the usage of nullable object type here to eliminate the many usage of `!!` force operator for converting nullable to its non-nullable equivalent here.

Reply:
Not quite follow,  do you mean  "var user_text = QuestionText?.text!!"?

Reply:
ah just a general question wondering:

```
    private var TitleText: TextView? = null
    private var ArticleText: TextView? = null
    private var AnswerText: TextView? = null
    private var QuestionText: TextView? = null
    private var DoQAButton: Button? = null
```
is there anything in this list can be made non-nullable? as force operator `!!` for converting is recommended to generally avoid unless there's a must.

Reply:
I see. Thanks. 'Lazy init' can make it.